### DISCUSSION TITLE
Increment column value on READ/SELECT

### DISCUSSION THREAD

**Discussion by rotimi-best** [supabase/supabase#2725] [Questions] [answered]
Is there a way to increment a column value every time I perform a read? My use case is this, I want to have a `view_count` column where I store how many times a particular `document` was viewed, just as youtube has a views.

I initial thought of using triggers just as is suggested for auto updating the `update_at` column, however I saw the triggers available has no event for select, just [insert, update and delete](https://www.postgresql.org/docs/current/sql-createtrigger.html). Is there a suggestion on how to auto increment this or implement this without have to manually make a different request to supabase to increment to `view_count`

**Comment by steve-chavez** [comment, marked as answer]
Hey @rotimi-best,

You can do this with a plpgsql function and call it through [RPC](https://supabase.io/docs/reference/javascript/rpc). Basically:

```sql
create or replace function read_update_document(doc_id int) returns setof documents as $$
begin
  update documents set view_count = view_count + 1 where id = doc_id;
  return query select * from documents where id = doc_id;
end;$$ language plpgsql volatile;
```

```js
const { data, error } = await supabase
  .rpc('read_update_document')
```
